File: Easy/896. Monotonic Array/896. Monotonic Array.ts
function isMonotonic(nums: number[]): boolean {
    let increasing = false;
    let decreasing = false;

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > nums[i - 1]) {
            increasing = true;
        } else if (nums[i] < nums[i - 1]) {
            decreasing = true;
        }

        if (increasing && decreasing) {
            // If we find both increasing and decreasing elements, the array is not monotonic.
            return false;
        }
    }

    return true;
}

File: Easy/896. Monotonic Array/896. Monotonic Array.rs
impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
        let n = nums.len();
        if n <= 1 {
            return true;
        }

        let mut increasing = true;
        let mut decreasing = true;

        for i in 1..n {
            if nums[i] > nums[i - 1] {
                decreasing = false;
            }
            if nums[i] < nums[i - 1] {
                increasing = false;
            }
        }

        increasing || decreasing
    }
}
